### PR TITLE
Bump github.com/Azure/go-autorest/autorest/adal to 0.9.5

### PR DESCRIPTION
**Pull request by markusthoemmes**
**What type of PR is this?**

**What this PR does / why we need it**:

Bumps the github.com/Azure/go-autorest/autorest/adal dependency to v0.9.5 to address https://snyk.io/vuln/SNYK-GOLANG-GITHUBCOMDGRIJALVAJWTGO-596515.

**Special notes for your reviewer**:

This doesn't completely remove the dependency in question from the codebase (yet). It's a first step that removes it from some parts of it though. Other steps can be contributed separately.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: vendor/github.com/form3tech-oss/jwt-go/claims.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+// if the token is invalid for any supported reason
+type Claims interface {
+	Valid() error
+}
+
+// Structured version of Claims Section, as referenced at
+// https://tools.ietf.org/html/rfc7519#section-4.1
+// See examples for how to use this with your own claim types
+type StandardClaims struct {
+	Audience  []string `json:"aud,omitempty"`
```
Comment (liggitt) [inline vendor/github.com/form3tech-oss/jwt-go/claims.go:19]: this changed the type from `string` to `[]string`... will that break compatibility with decoding any existing tokens?

Comment (markusthoemmes) [reply vendor/github.com/form3tech-oss/jwt-go/claims.go:19]: If Azure happily changed it in their codebase, I assumed things would be fine. :man_shrugging:.

**Pull request merged**